src/content/language/pattern-types.md

```diff
@@ -288,11 +288,10 @@ x || y && z => 'matches true',
 // ...
 ```
```

Comment:
This example is a little bit convoluted because `(x || y) && z` doesn't match any boolean values at all. E.g.,

```dart
const x = true;
const y = true;
const z = false;

int f(bool b) => switch (b) {
      x || y && z => 1, // Matches `true`.
      (x || y) && z => 2, // Does not match any `bool` value.
      _ => 3, // Matches every `bool` value.
    };

void main() {
  print(f(true)); // Prints '1'.
  print(f(false)); // Prints '3'.
}
```

We can't just consider the pattern `(x || y) && z` to be similar to the boolean expression `(x || y) && z`.

The expression evaluates to false because `(x || y)` evaluates to true and `z` evaluates to false, which is then the result of the `&&`.

However, the pattern `(x || y) && z` doesn't match true: `x` matches true, so we don't even try to match `y`, we know that `(x || y)` matches true. But `z` does not match true, so `(x || y) && z` doesn't match true.

Similarly, `(x || y) && z` doesn't match false, because `(x || y)` doesn't match false (and we don't even try matching with `z`).

This means that the strings in lines 286 and 287 are misleading.

Reply:
That's great, thanks!